### PR TITLE
Artifacts path update at right location

### PR DESCRIPTION
**Pull request by vibrantvarun**
### Description
The PR fixes the artifact publishing issue with job scheduler. I have added a new publication called mavenJava in the build .gradle which does packaging in Jar to the right publishing path.

Steps I followed to test:
1) I have changed repositories path from sonatype URL to my local machine
2) Added a new publishing end point mavenJava in build.gradle
3) Ran below command 
```
./gradlew opensearch-job-scheduler-spi:publishAllPublicationsToSnapshotsRepository && ./gradlew publishPluginZipPublicationToSnapshotsRepository && ./gradlew publishPluginJarToSnapshotsRepository
```

Please find the test output below.
```
.
└── org
    └── opensearch
        ├── opensearch-job-scheduler
        │   ├── 3.0.0.0-SNAPSHOT
        │   │   ├── maven-metadata.xml
        │   │   ├── maven-metadata.xml.md5
        │   │   ├── maven-metadata.xml.sha1
        │   │   ├── maven-metadata.xml.sha256
        │   │   ├── maven-metadata.xml.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-javadoc.jar
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-javadoc.jar.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-javadoc.jar.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-javadoc.jar.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-javadoc.jar.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-sources.jar
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-sources.jar.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-sources.jar.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-sources.jar.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1-sources.jar.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1.pom
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1.pom.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1.pom.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203646-1.pom.sha256
        │   │   └── opensearch-job-scheduler-3.0.0.0-20230503.203646-1.pom.sha512
        │   ├── maven-metadata.xml
        │   ├── maven-metadata.xml.md5
        │   ├── maven-metadata.xml.sha1
        │   ├── maven-metadata.xml.sha256
        │   └── maven-metadata.xml.sha512
        ├── opensearch-job-scheduler-spi
        │   ├── 3.0.0.0-SNAPSHOT
        │   │   ├── maven-metadata.xml
        │   │   ├── maven-metadata.xml.md5
        │   │   ├── maven-metadata.xml.sha1
        │   │   ├── maven-metadata.xml.sha256
        │   │   ├── maven-metadata.xml.sha512
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-javadoc.jar
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-javadoc.jar.md5
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-javadoc.jar.sha1
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-javadoc.jar.sha256
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-javadoc.jar.sha512
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-sources.jar
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-sources.jar.md5
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-sources.jar.sha1
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-sources.jar.sha256
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1-sources.jar.sha512
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.jar
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.jar.md5
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.jar.sha1
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.jar.sha256
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.jar.sha512
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.pom
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.pom.md5
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.pom.sha1
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.pom.sha256
        │   │   └── opensearch-job-scheduler-spi-3.0.0.0-20230503.203639-1.pom.sha512
        │   ├── maven-metadata.xml
        │   ├── maven-metadata.xml.md5
        │   ├── maven-metadata.xml.sha1
        │   ├── maven-metadata.xml.sha256
        │   └── maven-metadata.xml.sha512
        └── plugin
            └── opensearch-job-scheduler
                ├── 3.0.0.0-SNAPSHOT
                │   ├── maven-metadata.xml
                │   ├── maven-metadata.xml.md5
                │   ├── maven-metadata.xml.sha1
                │   ├── maven-metadata.xml.sha256
                │   ├── maven-metadata.xml.sha512
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.pom
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.pom.md5
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.pom.sha1
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.pom.sha256
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.pom.sha512
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.zip
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.zip.md5
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.zip.sha1
                │   ├── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.zip.sha256
                │   └── opensearch-job-scheduler-3.0.0.0-20230503.203644-1.zip.sha512
                ├── maven-metadata.xml
                ├── maven-metadata.xml.md5
                ├── maven-metadata.xml.sha1
                ├── maven-metadata.xml.sha256
                └── maven-metadata.xml.sha512
```
 
### Issues Resolved
[https://github.com/opensearch-project/job-scheduler/issues/374](https://github.com/opensearch-project/job-scheduler/issues/374)

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -136,11 +136,31 @@ publishing {
                 groupId = "org.opensearch.plugin"
             }
         }
+        mavenJava(MavenPublication){ publication ->
```
Comment (cwperks) [inline build.gradle:139]: nit: with the other publication called `pluginZip`, do you think we should call this `pluginJar`?

**Review by cwperks — COMMENTED** [review]
Great job figuring this out @vibrantvarun! I have checked out this branch and verified the changes locally.

File: build.gradle [hunk 2]
```diff
@@ -103,25 +103,25 @@ allprojects {
...
-                        Node license = node.appendNode('licenses').appendNode('license')
-                        license.appendNode('name', project.licenseName)
-                        license.appendNode('url', project.licenseUrl)
-
-                        Node developer = node.appendNode('developers').appendNode('developer')
-                        developer.appendNode('name', 'OpenSearch')
-                        developer.appendNode('url', 'https://github.com/opensearch-project/job-scheduler')
-                    }
-                }
+//               all {
```
[the remark this answers is not in the document]
Comment (vibrantvarun) [reply build.gradle:106]: Took care off

File: build.gradle [hunk 3]
```diff
@@ -136,11 +136,31 @@ publishing {
...
+                developer.appendNode('name', 'OpenSearch')
+                developer.appendNode('url', 'https://github.com/opensearch-project/job-scheduler')
+            }
+        }
     }
     repositories {
         maven {
             name = "Snapshots" //  optional target repository name
-            url = "https://aws.oss.sonatype.org/content/repositories/snapshots"
+            url = "https://aws.oss.sonatype.org/content/repositories/snapshots/snapshots"
```
[the remark this answers is not in the document]
Comment (vibrantvarun) [reply build.gradle]: Reverted it. Changed by mistake

File: build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (vibrantvarun) [reply build.gradle:106]: done

File: build.gradle (hunk 1 above)
Comment (vibrantvarun) [reply build.gradle]: done

File: build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (cwperks) [reply build.gradle:106]: I'm still seeing commented out lines in the PR.

[the remark this answers is not in the document]
Comment (saratvemulapalli) [reply build.gradle:106]: +1 lm still seeing commented code.

**Review by saratvemulapalli — COMMENTED** [review]
The changes look good to me. Thanks @vibrantvarun .
Could you take a look at the CI failures? 

@prudhvigodithi @gaiksaya how do we fix this for 2.7 published release after this change is merged?

**Comment by gaiksaya** [comment]
Hi @vibrantvarun,

Can you check if it publishing as expected on maven local too?
The reason being this script is called https://github.com/opensearch-project/job-scheduler/blob/main/scripts/build.sh when we build JS into distribution. AFAIK the components are logged into [build manifest ](https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.7.0/7771/linux/x64/tar/builds/opensearch/manifest.yml)as below:
<details><summary>Details</summary>
<p>

```
  - name: job-scheduler
    repository: https://github.com/opensearch-project/job-scheduler.git
    ref: e63a2e2ce9a55524bbc9eaef7c924cac7f3cbeea
    commit_id: e63a2e2ce9a55524bbc9eaef7c924cac7f3cbeea
    artifacts:
      maven:
        - maven/org/opensearch/plugin/opensearch-job-scheduler/maven-metadata.xml
        - maven/org/opensearch/plugin/opensearch-job-scheduler/maven-metadata.xml.sha1
        - maven/org/opensearch/plugin/opensearch-job-scheduler/maven-metadata.xml.md5
        - maven/org/opensearch/plugin/opensearch-job-scheduler/maven-metadata.xml.sha256
        - maven/org/opensearch/plugin/opensearch-job-scheduler/maven-metadata.xml.sha512
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.jar
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.jar.sha1
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.jar.md5
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.jar.sha256
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.jar.sha512
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.pom
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.pom.sha1
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.pom.md5
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.pom.sha256
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.pom.sha512
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.module
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.module.sha1
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.module.md5
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.module.sha256
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.module.sha512
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-javadoc.jar
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-javadoc.jar.sha1
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-javadoc.jar.md5
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-javadoc.jar.sha256
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-javadoc.jar.sha512
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-sources.jar
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-sources.jar.sha1
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-sources.jar.md5
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-sources.jar.sha256
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0-sources.jar.sha512
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.zip
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.zip.sha1
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.zip.md5
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.zip.sha256
        - maven/org/opensearch/plugin/opensearch-job-scheduler/2.7.0.0/opensearch-job-scheduler-2.7.0.0.zip.sha512
        - maven/org/opensearch/opensearch-job-scheduler-spi/maven-metadata.xml
        - maven/org/opensearch/opensearch-job-scheduler-spi/maven-metadata.xml.sha1
        - maven/org/opensearch/opensearch-job-scheduler-spi/maven-metadata.xml.md5
        - maven/org/opensearch/opensearch-job-scheduler-spi/maven-metadata.xml.sha256
        - maven/org/opensearch/opensearch-job-scheduler-spi/maven-metadata.xml.sha512
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.jar
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.jar.sha1
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.jar.md5
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.jar.sha256
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.jar.sha512
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.pom
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.pom.sha1
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.pom.md5
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.pom.sha256
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0.pom.sha512
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-sources.jar
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-sources.jar.sha1
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-sources.jar.md5
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-sources.jar.sha256
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-sources.jar.sha512
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-javadoc.jar
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-javadoc.jar.sha1
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-javadoc.jar.md5
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-javadoc.jar.sha256
        - maven/org/opensearch/opensearch-job-scheduler-spi/2.7.0.0/opensearch-job-scheduler-spi-2.7.0.0-javadoc.jar.sha512
      plugins:
        - plugins/opensearch-job-scheduler-2.7.0.0.zip
    version: 2.7.0.0
```

@saratvemulapalli @prudhvigodithi I believe we might need to document the process step by step of manual checking out the job scheduler repo at tag 2.7.0.0, cherry-pick this commit, get maven credentials and only run publish task for this URL: https://aws.oss.sonatype.org/content/repositories/releases/org/opensearch/opensearch-job-scheduler/

</p>
</details>

**Pull request closed without merging**